### PR TITLE
Fixed issue on rendering checkbox status of tree node

### PR DESCRIPTION
**Pull request by geleems**
* Issue: Checkbox status is always rendered as `-`

* Cause: `checked` and `enabled` field are missing when item created from tree data model for rendering.
Therefore, `checked` field always has `unidentified` value on rendering time.
(Missing `enabled` field did not cause issue since `unidentified` is considered as `true` for `enabled` field.)

Before fix:
image

After fix: 
image

### CODE REVIEW

File: src/vs/workbench/api/common/extHostTreeViews.ts [hunk 1]
```diff
@@ -467,6 +467,8 @@ export class ExtHostTreeView<T> extends Disposable {
 			tooltip: typeof extensionTreeItem.tooltip === 'string' ? extensionTreeItem.tooltip : undefined,
 			command: extensionTreeItem.command ? this.commands.toInternal(extensionTreeItem.command, disposable) : undefined,
 			contextValue: extensionTreeItem.contextValue,
+			enabled: (<any>element).enabled,
```
Comment (kevcunnane) [inline src/vs/workbench/api/common/extHostTreeViews.ts]: I think we need a SQL Carbon Edit to identify this - there's one lower in the `payload:` part so maybe we can add one here and then do `// {{SQL CARBON EDIT}} - End` after the  end of this `const item` definition?

**Comment by Charles-Gagnon** [comment]
> 		const disposable = new DisposableStore();

This is overridden in [https://github.com/microsoft/azuredatastudio/blob/c23738f9355b5e759b54102400c58af499eae678/src/sql/workbench/api/common/extHostModelViewTree.ts#L163](https://github.com/microsoft/azuredatastudio/blob/c23738f9355b5e759b54102400c58af499eae678/src/sql/workbench/api/common/extHostModelViewTree.ts#L163) which adds in the enabled and checked properties - would that not be the appropriate place to do this? That method is pulling those properties from extensionTreeItem which is different than where you're getting it here so what caused the change and do we need to update that logic too in addition if this is still necessary for some reason?

---
Refers to: src/vs/workbench/api/common/extHostTreeViews.ts:458 in c357ef6. [](commit_id = c357ef6a901a37ce12d8d5eb46ec4d505540883a, deletion_comment = False)

**Comment by geleems** [comment]
It turned out the return value of `exHostTreeView.createTreeNode()` was changed 4 days ago, and it made `exHostModelViewTree.createTreeNode()` add those 2 field to wrong place. 
image
I will update `exHostModelViewTree.createTreeNode()` to add up those 2 fields to correct place

File: src/sql/workbench/api/common/extHostModelViewTree.ts [hunk 2]
```diff
@@ -161,8 +161,10 @@ export class ExtHostTreeView<T> extends vsTreeExt.ExtHostTreeView<T> {
 	}
 
 	protected createTreeNode(element: T, extensionTreeItem: azdata.TreeComponentItem, parent?: vsTreeExt.TreeNode): vsTreeExt.TreeNode {
-		let item = super.createTreeNode(element, extensionTreeItem, parent);
-		item = Object.assign({}, item, { checked: extensionTreeItem.checked, enabled: extensionTreeItem.enabled });
-		return item;
+		let node = super.createTreeNode(element, extensionTreeItem, parent);
+		if (node.item) {
+			node.item = Object.assign(node.item, { checked: extensionTreeItem.checked, enabled: extensionTreeItem.enabled });
```
Comment (kevcunnane) [inline src/sql/workbench/api/common/extHostModelViewTree.ts]: Should we just cast as an azdata.TreeComponentItem and then directly assign these values?

Comment (geleems) [reply src/sql/workbench/api/common/extHostModelViewTree.ts]: the item is in `ITreeItem` type, and casting to `azdata.TreeComponentItem` fails.
I can directly assign by `node.item['checked'] = extensionTreeItem.checked` though if you want.

**Pull request merged**